### ISSUE TITLE
visit-install error installing on mac

### ISSUE THREAD

**Issue by aowen87** [aowen87/BAR#997]
The visit-install script has some erroneous logic. Below is an e-mail describing it with a solution.

Hello,

I've been trying to install Visit on my macbook pro core duo (32bits)
/ mac OS 10.6.6, using the visit-install script, as written in the
readme file.

I do this :

./visit-install 2.2.1 darwin-i386 ~/visit

the script prints some stuff in the terminal and finishes with this :

667495 blocks
64426 blocks
find: /Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin//Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin/gui.app/Contents/MacOS:
No such file or directory
find: /Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin//Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin/mcurvit.app/Contents/MacOS:
No such file or directory
find: /Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin//Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin/viewer.app/Contents/MacOS:
No such file or directory
find: /Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin//Users/nicolasaunai/visit/2.2.1+/darwin-i386/bin/xmledit.app/Contents/MacOS:
No such file or directory
visit-install done

The version I've downloaded is :
https://wci.llnl.gov/codes/visit/2.2.1/VisIt-2.2.1-i386-installer.dmg

I think the error comes from the following part of the visit-install code :

if [ "$os" = "darwin" ]; then
    for arch in `ls "$dir"/$version+`; do
        if [ "$arch" != "bin" ]; then
            for app in `find "$dir"/$version+/$arch/bin -name "*.app"
-print`; do
                find "$dir"/$version+/$arch/bin/$app/Contents/MacOS
-exec chmod $dir_permission {} \;
            done
        fi
    done
fi

the $app variable is already a complete absolute path, not the name of
the .app file as the find command seems to be written for...

The error disappears if the previous lines are replaced by the following :

if [ "$os" = "darwin" ]; then
    for arch in `ls "$dir"/$version`; do
        if [ "$arch" != "bin" ]; then
            for app in `find "$dir"/$version/$arch/bin -name "*.app" -print`; do
                find $app/Contents/MacOS -exec chmod $dir_permission {} \;
            done
        fi
    done
fi

Unfortunatly, even with this correction, I can't make visit work :

If I click on the 'gui.app' in the finder in
visit_directory/2.2.1/darwin-i386/bin/gui.app

the application is launch but seems not responding before starting...
(blocked at the initialization window at the begining of the progress
bar)

if I run the 'visit' script in the terminal :

[~/visit/bin]$ ./visit
Version 2.2.1 of 'gui' does not exist for
the architecture 'darwin-x86_64'.
VisIt located binaries for the following architectures:
        darwin-i386

What should I do ?

Thanks
Nico

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Eric Brugger
Original creation: 03/07/2011 08:01 pm
Original update: 03/21/2011 12:36 pm
Ticket number: 644
